Clean up and extend the Widget coordinate conversion functions

smfr opened the pull request:
#### e50c61a139e343cc6e6611510f6490e05b12e3c5
<pre>
Clean up and extend the Widget coordinate conversion functions
<a href="https://bugs.webkit.org/show_bug.cgi?id=294530">https://bugs.webkit.org/show_bug.cgi?id=294530</a>
<a href="https://rdar.apple.com/153473471">rdar://153473471</a>

Reviewed by Alan Baradlay.

A future change will require `convertFromRootToContainingWindow()` and `convertFromContainingWindowToRoot()`
which support FloatPoint, so add `Widget::convert*` variants for FloatPoint/FloatRect.

Also clean up the declarations and the definitions; order consistently by function then type
(IntPoint, FloatPoint, IntRect, FloatRect).

Pass IntPoint and FloatPoint arguments by value.

* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::convertFromRendererToContainingView const):
(WebCore::FrameView::convertFromContainingViewToRenderer const):
(WebCore::FrameView::convertToContainingView const):
(WebCore::FrameView::convertFromContainingView const):
* Source/WebCore/page/FrameView.h:
* Source/WebCore/platform/ScrollView.cpp:
(WebCore::ScrollView::viewToContents const):
(WebCore::ScrollView::windowToContents const):
(WebCore::ScrollView::contentsToWindow const):
(WebCore::ScrollView::convertSelfToChild const):
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/Scrollbar.cpp:
(WebCore::Scrollbar::convertToContainingView const):
(WebCore::Scrollbar::convertFromContainingView const):
* Source/WebCore/platform/Scrollbar.h:
* Source/WebCore/platform/Widget.cpp:
(WebCore::Widget::convertToRootView const):
(WebCore::Widget::convertFromRootView const):
(WebCore::Widget::convertToContainingWindow const):
(WebCore::Widget::convertFromContainingWindow const):
(WebCore::Widget::convertToContainingView const):
(WebCore::Widget::convertFromContainingView const):
(WebCore::Widget::convertFromRootToContainingWindow):
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/platform/Widget.h:
* Source/WebCore/platform/ios/WidgetIOS.mm:
(WebCore::Widget::convertFromRootToContainingWindow):
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/platform/mac/WidgetMac.mm:
(WebCore::Widget::convertFromRootToContainingWindow):
(WebCore::Widget::convertFromContainingWindowToRoot):

Canonical link: <a href="https://commits.webkit.org/296272@main">https://commits.webkit.org/296272@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/89f0185da6690d305b6b2fa480fa949802de21a0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108006 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27671 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18090 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113217 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58526 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109969 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28366 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36220 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81994 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110954 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22482 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97309 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62425 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21894 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15442 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57964 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91835 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15508 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116344 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35078 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25822 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91025 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35454 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93587 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90819 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23141 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35717 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13474 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/30889 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34976 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40530 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34720 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38079 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36380 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/page/FrameView.cpp:
```diff
@@ -191,16 +191,63 @@ bool FrameView::formControlRefreshEnabled() const
+
+IntPoint FrameView::convertFromRendererToContainingView(const RenderElement* renderer, IntPoint rendererPoint) const
+{
+    IntPoint point = roundedIntPoint(renderer->localToAbsolute(rendererPoint, UseTransforms));
```
```suggestion
    auto point = roundedIntPoint(renderer->localToAbsolute(rendererPoint, UseTransforms));
```

alanbaradlay reviewed (approved):
`// MARK: -`

shouldn't this say something to be meaningful?

smfr committed:
Clean up and extend the Widget coordinate conversion functions
https://bugs.webkit.org/show_bug.cgi?id=294530
rdar://153473471

Reviewed by Alan Baradlay.

A future change will require `convertFromRootToContainingWindow()` and `convertFromContainingWindowToRoot()`
which support FloatPoint, so add `Widget::convert*` variants for FloatPoint/FloatRect.

Also clean up the declarations and the definitions; order consistently by function then type
(IntPoint, FloatPoint, IntRect, FloatRect).

Pass IntPoint and FloatPoint arguments by value.

* Source/WebCore/page/FrameView.cpp:
(WebCore::FrameView::convertFromRendererToContainingView const):
(WebCore::FrameView::convertFromContainingViewToRenderer const):
(WebCore::FrameView::convertToContainingView const):
(WebCore::FrameView::convertFromContainingView const):
* Source/WebCore/page/FrameView.h:
* Source/WebCore/platform/ScrollView.cpp:
(WebCore::ScrollView::viewToContents const):
(WebCore::ScrollView::windowToContents const):
(WebCore::ScrollView::contentsToWindow const):
(WebCore::ScrollView::convertSelfToChild const):
* Source/WebCore/platform/ScrollView.h:
* Source/WebCore/platform/Scrollbar.cpp:
(WebCore::Scrollbar::convertToContainingView const):
(WebCore::Scrollbar::convertFromContainingView const):
* Source/WebCore/platform/Scrollbar.h:
* Source/WebCore/platform/Widget.cpp:
(WebCore::Widget::convertToRootView const):
(WebCore::Widget::convertFromRootView const):
(WebCore::Widget::convertToContainingWindow const):
(WebCore::Widget::convertFromContainingWindow const):
(WebCore::Widget::convertToContainingView const):
(WebCore::Widget::convertFromContainingView const):
(WebCore::Widget::convertFromRootToContainingWindow):
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/platform/Widget.h:
* Source/WebCore/platform/ios/WidgetIOS.mm:
(WebCore::Widget::convertFromRootToContainingWindow):
(WebCore::Widget::convertFromContainingWindowToRoot):
* Source/WebCore/platform/mac/WidgetMac.mm:
(WebCore::Widget::convertFromRootToContainingWindow):
(WebCore::Widget::convertFromContainingWindowToRoot):

Canonical link: https://commits.webkit.org/296272@main
